engines/ic3ia.h

```diff
@@ -19,57 +19,90 @@
 **        and the open source implementation:
 **
 **        https://es-static.fbk.eu/people/griggio/ic3ia/index.html
+**
+**  within Pono, we are building on the bit-level IC3 instead of directly
+**  on IC3Base, because a lot of the functionality is the same
+**  In particular, we don't need to override either of the generalization
+**  functions. Instead focusing on abstract/refine.
+**
 **/
 
-#include "engines/mbic3.h"
+#pragma once
+
+#include "engines/ic3.h"
 #include "modifiers/implicit_predicate_abstractor.h"
+#include "smt-switch/term_translator.h"
 
 namespace pono {
 
-// process is mostly the same as model based IC3,
-// but overrides a few methods to only use cubes/clauses over predicates
-class IC3IA : public ModelBasedIC3
+class IC3IA : public IC3
 {
  public:
-  IC3IA(Property & p, smt::SolverEnum se);
-  IC3IA(Property & p, const smt::SmtSolver & slv);
-  IC3IA(const PonoOptions & opt, Property & p, smt::SolverEnum se);
-  IC3IA(const PonoOptions & opt, Property & p, const smt::SmtSolver & slv);
-  virtual ~IC3IA();
-
-  typedef ModelBasedIC3 super;
+  // itp_se is the SolverEnum for the interpolator
+
+  IC3IA(Property & p,
+        smt::SolverEnum se,
+        smt::SolverEnum itp_se = smt::SolverEnum::MSAT);
+  IC3IA(Property & p,
+        const smt::SmtSolver & s,
+        smt::SolverEnum itp_se = smt::SolverEnum::MSAT);
+  IC3IA(const PonoOptions & opt,
+        Property & p,
+        smt::SolverEnum se,
+        smt::SolverEnum itp_se = smt::SolverEnum::MSAT);
+  IC3IA(const PonoOptions & opt,
```

Comment:
empty line space

Reply:
done in 776b789b56bedbe54bb965348ada5bd2621dd658